app-android/src/main/java/com/ustadmobile/port/android/view/ContentEntryEdit2Fragment.kt

```diff
@@ -86,6 +122,53 @@ class ContentEntryEdit2Fragment(private val registry: ActivityResultRegistry? =
             field = value
         }
 
+    override var videoFilePath: String? = null
+        get() = field
+        set(value) {
+
+            field = value
+            if(value == null) return
+            if (value.startsWith("http")) {
+                mBinding?.showVideoPreview = false
+                mBinding?.showWebPreview = true
+                prepareWeb(value)
+            }else{
+                mBinding?.showVideoPreview = true
+                mBinding?.showWebPreview = false
+                prepareVideo(value)
+            }
+        }
+
+    fun prepareVideo(filePath: String) {
+        val uri = Uri.parse(filePath)
+        val dataSourceFactory: DataSource.Factory = DefaultDataSourceFactory(requireContext(), "UstadMobile")
+        val mediaSource = ProgressiveMediaSource.Factory(dataSourceFactory)
+                .createMediaSource(uri)
+        player?.prepare(mediaSource)
+    }
+
+    fun prepareWeb(filePath: String){
+        webView?.loadData("""
+            <!DOCTYPE html>
+            <html lang="en">
+            <body>
+            
+             <video id="video" width="100%" height="175px" controls>
```

Comment:
We don't know for sure that this is an mp4. As there is only one source, you don't need to specify the type, e.g.
```
<video src="$filePath" width="100%" height="175px" controls></video>
```

Reply:
Also css style attribute should be used for setting width and height.

e.g. style='width: 100%; height: 175px'

Reply:
done